### ISSUE TITLE
How do you run the generated linux output?

### ISSUE THREAD

**Issue by grofit** [maxogden/electron-packager#115]
I apologise if this is a simple question, I rarely use Linux but wanted to add support for it as a platform.

Now I am using electron version 0.30.2 and the package generated contains all the `*.so` files and the main `Electron` file, but when I put them on linux (I installed ubuntu on a VM) it doesnt seem to know how to open the `Electron` file.

There seems to be very little information online around this, and I know that this library basically gets the electron files and just copies and renames them but just wanted to see if there was some common way to just run these outputs on a fresh linux install. I have seen references saying there should be an `Electron.app` which there isnt, and some snippets here and there mention python bootstraps but I cannot find any solid information on how to build in windows and deploy on linux anywhere... which is baffling as I assume this would be a common thing.

Any help/advice would be great

**Comment by malept** [comment]
`Electron.app` is for OSX.

To run the app in Linux:
1. Open up a terminal, `cd` to the output directory of your app.
2. Run: `./Electron` (though, I think the name of the executable should have changed to `$appname`.)

**Comment by grofit** [comment]
I tried doing this but I just get "Command not found", in reality the actual output is `MyAppName` not `Electron` but if I were to do `./MyAppName` it just gives the error above.

**Comment by malept** [comment]
Does `MyAppName` exist in the output directory?

**Comment by grofit** [comment]
Yeah, Here is a screenshot:

https://dl.dropboxusercontent.com/u/98595151/ShareX/EMBERSOFT1/electron-linux.jpg

**Comment by kfranqueiro** [comment]
What platform did you actually run `electron-packager` on?  IIRC, Windows eats the proper permissioning of the Linux executable which should be executable.  If you do `ls -l` and see no `x` bits on `MyAppName`, that's your problem.

**Comment by grofit** [comment]
I ran it via grunt with these settings:

```
electron: {
    windows: {
        // Lets not care about this
    },
    linux: {
        options: {
            dir: "./app",
            name: "MyAppName",
            platform: "linux",
            arch: "all",
            asar: true,
            overwrite: true,
            version: "0.30.2",
            icon: "resources/my-app-icon.ico",
            out: "_packages/raw",
            "app-version": appPkg.version   // appPkg is just a reference to the package.json
        }
    }
},
```

Using windows 10, latest Grunt (as of like a week ago) and latest version of this lib.

**Comment by grofit** [comment]
HUZZAH! ok so if I right click there is a check box "Allow Running As Executable", clicked that and boom runs like it should do.

So I guess the next question is as briefly mentioned above, is there a way to preset these permissions?

**Comment by malept** [comment]
You might have to build your app inside a Linux VM to do so. (You can build Electron-based apps for all platforms on Linux.)

**Comment by grofit** [comment]
Yeah, I just didnt want to have to start hopping between O/S to build stuff, although I heard somewhere that I would need to do that for Mac OS anyway, as apparently you can only sign stuff with XCode.

**Comment by heruputra** [comment]
Any update of this?

**Comment by malept** [comment]
I don't know how to fix this on Windows hosts. This would be a good bug for someone in the community to research and see if a fix is even possible.

**Comment by MarshallOfSound** [comment]
This isn't something Packager can fix. On Windows there's simply nowhere to store a POSIX execute bit, NTFS doesn't have one, so the permission is lost at extraction time, and even if we could restore it, copying a plain directory from a Windows filesystem to Linux would strip it again. Setting the exec bit correctly is the job of whatever produces the final distributable (an archive/installer tool that writes unix modes, e.g. Forge's makers), or of building on a POSIX host (Linux, macOS, WSL, or CI).

Closing as this isn't fixable within Packager itself.
